src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3379,87 +3352,84 @@ void EmitLazy(RegexNode node)
                 Goto(originalDoneLabel);
                 writer.WriteLine();
 
-                MarkLabel(endLoop, emitSemicolon: isAtomic);
+                MarkLabel(endLoop, emitSemicolon: false);
 
-                // If the lazy loop is not atomic, then subsequent code may backtrack back into this lazy loop, either
+                // The lazy loop is not atomic, so subsequent code may backtrack back into this lazy loop, either
```

Comment:
I'm not following why can we now guarantee that this is not atomic now and don't need to check anymore. How does this work now for non-atomic lazy loops?

Reply:
If everything is working correctly and the analysis phase is correctly lowering the upper bound to the lower bound for atomic loops, it will be caught by the `if (minIterations == maxIterations)` check earlier and end up delegating to EmitLoop instead, and we then assert that !IsAtomicByAncestor.  If we mess up our analysis and don't lower the bound, we could still reach here with an atomic lazy loop.  I can revert this comment, though, and put back the atomic check below, just in case.